File: library/src/main/java/org/incendo/interfaces/inventory/ChestInterfacesInventory.java
package org.incendo.interfaces.inventory;

import net.kyori.adventure.text.Component;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.InventoryHolder;
import org.bukkit.inventory.ItemStack;
import org.incendo.interfaces.utilities.BukkitInventoryUtilities;

public final class ChestInterfacesInventory extends CachedInterfacesInventory {

    private final Inventory chestInventory;

    public ChestInterfacesInventory(final InventoryHolder holder, final Component title, final int rows) {
        this.chestInventory = BukkitInventoryUtilities.createBukkitInventory(holder, rows, title);
    }

    public Inventory chestInventory() {
        return this.chestInventory;
    }

    @Override
    public ItemStack get(final int row, final int column) {
        int index = BukkitInventoryUtilities.gridPointToBukkitIndex(row, column);
        return this.chestInventory.getItem(index);
    }

    @Override
    public void setInternal(final int row, final int column, final ItemStack item) {
        int index = BukkitInventoryUtilities.gridPointToBukkitIndex(row, column);
        this.chestInventory.setItem(index, item);
    }

    @Override
    public boolean isPlayerInventory(final int row, final int column) {
        return false;
    }
}

File: next/src/main/kotlin/org/incendo/interfaces/next/inventory/ChestInterfacesInventory.kt
package org.incendo.interfaces.next.inventory

import net.kyori.adventure.text.Component
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.InventoryHolder
import org.bukkit.inventory.ItemStack
import org.incendo.interfaces.next.utilities.createBukkitInventory
import org.incendo.interfaces.next.utilities.gridPointToBukkitIndex

public class ChestInterfacesInventory(
    holder: InventoryHolder,
    title: Component?,
    rows: Int,
) : CachedInterfacesInventory() {

    public val chestInventory: Inventory = createBukkitInventory(holder, rows, title)

    override fun get(row: Int, column: Int): ItemStack? {
        val index = gridPointToBukkitIndex(row, column)
        return chestInventory.getItem(index)
    }

    override fun setInternal(row: Int, column: Int, item: ItemStack?) {
        val index = gridPointToBukkitIndex(row, column)
        chestInventory.setItem(index, item)
    }
}
